src/librssguard/gui/toolbars/messagestoolbar.cpp

```diff
@@ -125,26 +140,128 @@ void MessagesToolBar::initializeSearchBox() {
+    "filter_last48hours");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from this week"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterThisWeek),
+    "filter_this_week");
+  addActionToMenu(m_menuMessageFilter,
+    qApp->icons()->fromTheme(QSL("mail-mark-read")),
+    tr("Filter articles from last week"),
+    QVariant::fromValue(MessagesProxyModel::MessageFilter::FilterLastWeek),
+    "filter_last_week");
+
   m_btnMessageHighlighter = new QToolButton(this);
   m_btnMessageHighlighter->setToolTip(tr("Display all articles"));
   m_btnMessageHighlighter->setMenu(m_menuMessageHighlighter);
   m_btnMessageHighlighter->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
   m_btnMessageHighlighter->setIcon(qApp->icons()->fromTheme(QSL("mail-mark-read")));
+  m_btnMessageHighlighter->setDefaultAction(m_menuMessageHighlighter->actions().front());
+  m_btnMessageFilter = new QToolButton(this);
+  m_btnMessageFilter->setToolTip(tr("Display all articles"));
+  m_btnMessageFilter->setMenu(m_menuMessageFilter);
+  m_btnMessageFilter->setPopupMode(QToolButton::ToolButtonPopupMode::MenuButtonPopup);
+  m_btnMessageFilter->setIcon(qApp->icons()->fromTheme(QSL("mail-mark-read")));
+  m_btnMessageFilter->setDefaultAction(m_menuMessageFilter->actions().front());
   m_actionMessageHighlighter = new QWidgetAction(this);
   m_actionMessageHighlighter->setDefaultWidget(m_btnMessageHighlighter);
   m_actionMessageHighlighter->setIcon(m_btnMessageHighlighter->icon());
   m_actionMessageHighlighter->setProperty("type", HIGHLIGHTER_ACTION_NAME);
   m_actionMessageHighlighter->setProperty("name", tr("Article highlighter"));
+  m_actionMessageFilter = new QWidgetAction(this);
+  m_actionMessageFilter->setDefaultWidget(m_btnMessageFilter);
+  m_actionMessageFilter->setIcon(m_btnMessageFilter->icon());
+  m_actionMessageFilter->setProperty("type", FILTER_ACTION_NAME);
+  m_actionMessageFilter->setProperty("name", tr("Article filter"));
 
   connect(m_menuMessageHighlighter, &QMenu::triggered, this, &MessagesToolBar::handleMessageHighlighterChange);
+  connect(m_menuMessageFilter, &QMenu::triggered, this, &MessagesToolBar::handleMessageFilterChange);
+}
+
+void MessagesToolBar::saveToolButtonSelection(const QString& buttonName, const QAction* action) const {
+  QStringList action_names = savedActions();
+  for (QString& action_name : action_names) {
+    if (action_name.startsWith(buttonName)) {
+      action_name = buttonName + (action->objectName().isEmpty() ? "" : "[" + action->objectName().toStdString() + "]").c_str();
+    }
+  }
+  qApp->settings()->setValue(GROUP(GUI), GUI::MessagesToolbarDefaultButtons, action_names.join(QSL(",")));
+}
+
+void MessagesToolBar::activateAction(const QString& action_name, QWidgetAction* widgetAction) {
+  const int start = action_name.indexOf('[');
+  const int end = action_name.indexOf(']');
```

Comment:
I format my code with "uncrustify" with config found in [uncrustify.cfg](https://github.com/martinrotter/rssguard/blob/master/resources/scripts/uncrustify/uncrustify.cfg)

Reply:
I will check if clang-format is good for me and will switch as it is indeed much more used than uncrustify these times.

Reply:
OK, I switched to clang-format.

https://github.com/martinrotter/rssguard/blob/master/.clang-format

I will gradually update the config file to make it work for me and will re-format all source files with it eventually.